src/account-info/AccountDetails.js

```diff
@@ -40,7 +40,7 @@ export default class AccountDetails extends PureComponent<Props, void> {
     return (
       <View>
         <Avatar
-          avatarUrl={getMediumAvatar(user.avatar_url)}
+          avatarUrl={user.avatar_url ? getMediumAvatar(user.avatar_url) : null}
```

Comment:
Hmmm, that doesn't look like the correct behavior.
If `avatar_url` is `null` then we should return a differently sized gravatar.

We might want to handle that inside `getMediumAvatar`

Reply:
So in `Avatar` we have this line:

```js
const fullAvatarUrl = avatarUrl ? getFullUrl(avatarUrl, realm) : getGravatarFromEmail(email);
```

So the avatar item explicitly accepts avatarUrls set to `null` and handles them by creating a gravatar. Isn't that the behavior that we want?

Reply:
Seems reasonable to me.

As a separate refactor, it feels like the question of whether the avatar should be "medium" or what should be determined in the same place as the size of the gravatar -- which means that `getMediumAvatar`, which is for the case of a non-null `avatar_url`, and `getGravatarFromEmail` should be called from the same spot in the code. Probably inside `Avatar`, especially as we're already giving it a `size` prop.

That's orthogonal from this PR, though, I think. (Basically, the reason anything works now before this change is just that `getMediumAvatar`, given null input, happens to return the same thing for its output; this at least makes the behavior in the null case clearer.)